CHANGELOG.md

```diff
@@ -7,6 +7,23 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [Unreleased]
 ### Added
 - `#![no_std]` is now supported out of the box. (You don’t need to opt into any features, it just works.)
+- [**BREAKING**] a `default_code` expression can now refer to the values of earlier fields by name.
+  (This is extremely unlikely to break your code, but could in theory due to shadowing.)
+- `#[builder(exclude)]` on fields, to not provide a method to set that field.
+- Control of documentation:
+  - `#[builder(doc = "…")]` on fields, to document the field’s method on the builder. Unlike `#[doc]`, you can currently only have one value rather than one attribute per line; but that’s not a big deal since you don’t get to use the `///` sugar anyway. Just use a multiline string.
+  - `#[builder(doc, builder_method_doc = "…", builder_type_doc = "…", build_method_doc = "…")]` on structs:
+    - `doc` unhides the builder type from the documentation.
+	- `builder_method_doc = "…"` replaces the default documentation that will be generated for the builder() method of the type for which the builder is being generated.
+	- `builder_type_doc = "…"` replaces the default documentation that will be generated for the builder type. Implies `doc`.
+	- `build_method_doc = "…"` replaces the default documentation that will be generated for the build() method of the builder type. Implies `doc`.
+
+### Changed
+- [**BREAKING**] Renamed the generated builder type from `TypedBuilder_BuilderFor_Foo` to `FooBuilder`, for improved ergonomics, especially when you enable documentation of the builder type. You can also now change it to something else with `#[builder(name = SomethingElse)]` on the type you are deriving TypedBuilder on.
+  - Generic identifiers were also changed, from `TypedBuilder_genericType_x` to `__x`. This is still expected to avoid all name collisions, but is easier to read in the builder type docs if you enable them.
```

Comment:
How is `__x` avoiding all name collisions?

Reply:
Prefixing a generic with double underscore is an existing convention for avoiding collisions with other generics. For example, Serde Serialize and Deserialize deriving use it (`__S`, `__D`, used on some other values that I can’t recall off the top of my head too). User code should not be using double underscore prefixes on anything ever, so collisions are expected to be avoided.

Reply:
And collisions with identifiers created by other proc-macros?

Reply:
The generics are only on the `FooBuilder` type, not the `Foo` type. You can’t `#[derive]` things or put custom attributes on the builder, so I’m content to say the practical scope for collision is nil. (Hmm, I wonder whether `#[builder_attr(derive(Clone))]` putting `#[derive(Clone)]` on the builder type would allow it to work. Should experiment with that before committing to this.)